analysisCode/src/TruthEvent.cpp

```diff
@@ -1,8 +1,82 @@
 #include "include/TruthEvent.h"
-
 #include <iostream>
 
-TruthEvent::TruthEvent()
+TruthEvent::TruthEvent(){cout << "A truth event" << endl;}
+
+void TruthEvent::ProcessEvent(erhic::EventPythia* ev, JetDef def, SoftDropJetDef softdrop_def)
+{ 
+  contrib::SoftDrop softdrop = softdrop_def.getSoftDrop(); 
+  Set_truthEvent(ev);
+  //PrintEvent();
+  ParticleLoop(ev);
+  JetLoop(def, softdrop);   
+ }
+
+void TruthEvent::ParticleLoop(erhic::EventPythia* ev) 
 {
-  std::cout<<"A truth event"<<std::endl;
+  // initialize variables for boost to Breit frame -- this will be moved
+  double Px,Py,Pz,P0,qx,qy,qz,q0;
+  for (int part = 0; part < nTracks; part++)
+    {
+      const Particle *truthPart = ev->GetTrack(part);
+      Set_truthPart(truthPart);
```

Comment:
Same comment as in the header file - I don't think this is needed here because we don't need to keep the particle information. We should just create the particle level variables that are needed locally, then pass them to the pseudojet vector, and then they will be discarded at the end of the function.

Reply:
I will get rid of "Set_truthPart" here. To clarify, the ParticleLoop function should still exist, just just as a way to fill a vector of Pseudojets, right?